src/core/geometry/qgsrectangle.cpp

```diff
@@ -145,8 +145,8 @@ QString QgsRectangle::toString( int precision ) const
     }
   }
 
-  if ( isEmpty() )
-    rep = QStringLiteral( "Empty" );
+  if ( isNull() )
```

Comment:
Shouldn't we still handle the `isEmpty` case here?

Reply:
I don't think so, or at least it's not specifically useful as the definition of an "empty" rectangle is just that the rectangle has no area (collapsed to a point or vertical or horizontal line)

Reply:
Of course this is being a problem anyway with null... eh:
https://github.com/qgis/QGIS/actions/runs/6429563312/job/17464475280?pr=54646#step:13:540

Reply:
But if QGIS has two different concepts for "null" and "empty" rectangles, shouldn't it be visible in the `toString` output? :shrug:

Reply:
Empty is not strictly a concept, more of a characteristic. It's like saying area is zero. The coordinates give the whole detail, there's no need for special handling because there isn't the problem of having to represent NaN numbers or max_double or similar when a supposedly common "null" rectangle is encountered.